### PR TITLE
Go uses FileHeaderView

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/ImportTypeTransformer.java [hunk 1]
```diff
@@ -24,12 +24,13 @@
 import java.util.Set;
 import java.util.TreeSet;
 
-public class ImportTypeTransformer {
+public class ImportTypeTransformer implements ImportGenerator {
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/ImportTypeTransformer.java]: I don't think we should have separate "transformer" and "generator" terms. I would suggest renaming this existing class `ImportTypeTransformer` to `StandardImportTypeTransformer` and renaming the interface `ImportGenerator` to `ImportTypeTransformer`.

File: src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicClientTransformer.java [hunk 2]
```diff
@@ -133,7 +127,7 @@ private StaticLangApiAndSettingsFileView generateApiAndSettingsView(
 
     // must be done as the last step to catch all imports
     csharpCommonTransformer.addCommonImports(context);
-    fileView.fileHeader(fileHeaderTransformer.generateFileHeader(context));
+    fileView.fileHeader(fileHeaderTransformer.generateFileHeader(context, importTypeTransformer));
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicClientTransformer.java]: I would prefer the fileHeaderTransformer to take importTypeTransformer in its constructor.

File: src/main/java/com/google/api/codegen/transformer/ImportTypeTransformer.java (hunk 1 above)
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/ImportTypeTransformer.java]: Done

File: src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicClientTransformer.java (hunk 2 above)
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicClientTransformer.java]: Done

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**